code/game/atoms.dm

```diff
@@ -2103,15 +2103,53 @@
 	SHOULD_NOT_SLEEP(TRUE)
 
 	var/mob/user = client?.mob
+	if (isnull(user))
+		return
 
 	// Screentips
-	var/datum/hud/active_hud = user?.hud_used
+	var/datum/hud/active_hud = user.hud_used
 	if(active_hud)
-		if(!active_hud.screentips_enabled || (flags_1 & NO_SCREENTIPS_1))
+		var/screentips_enabled = active_hud.screentips_enabled
+		if(screentips_enabled == SCREENTIP_PREFERENCE_DISABLED || (flags_1 & NO_SCREENTIPS_1))
 			active_hud.screentip_text.maptext = ""
 		else
-			//We inline a MAPTEXT() here, because there's no good way to statically add to a string like this
-			active_hud.screentip_text.maptext = "<span class='maptext' style='text-align: center; font-size: 32px; color: [active_hud.screentip_color]'>[name]</span>"
+			var/extra_context = ""
+
+			if (isliving(user))
+				var/obj/item/held_item = user.get_active_held_item()
+
+				if ((flags_1 & HAS_CONTEXTUAL_SCREENTIPS_1) || (held_item?.item_flags & ITEM_HAS_CONTEXTUAL_SCREENTIPS))
+					var/list/context = list()
+
+					var/contextual_screentip_returns = \
+						SEND_SIGNAL(src, COMSIG_ATOM_REQUESTING_CONTEXT_FROM_ITEM, context, held_item, user) \
+						| (held_item && SEND_SIGNAL(held_item, COMSIG_ITEM_REQUESTING_CONTEXT_FOR_TARGET, context, src, user))
+
+					if (contextual_screentip_returns & CONTEXTUAL_SCREENTIP_SET)
+						// LMB and RMB on one line...
+						var/lmb_text = (SCREENTIP_CONTEXT_LMB in context) ? "[SCREENTIP_CONTEXT_LMB]: [context[SCREENTIP_CONTEXT_LMB]]" : ""
+						var/rmb_text = (SCREENTIP_CONTEXT_RMB in context) ? "[SCREENTIP_CONTEXT_RMB]: [context[SCREENTIP_CONTEXT_RMB]]" : ""
+
```

Comment:
I was going to ask why you're doing this here rather then in the element, but I assume it's to support people going off script with things and it still looking nice

Reply:
Because there is no "the element", there's a lot of different ways to make screentips

Reply:
ye